sympy/stats/stochastic_process_types.py

```diff
@@ -0,0 +1,230 @@
+from sympy import (Symbol, Matrix, MatrixSymbol, S, Indexed, Basic,
+                    Set, And, Tuple, Eq, FiniteSet, ImmutableMatrix)
+from sympy.core.compatibility import string_types
+from sympy.stats.rv import RandomIndexedSymbol, random_symbols, RandomSymbol
+from sympy.stats.symbolic_probability import Probability
+from sympy.stats.stochastic_process import StochasticPSpace
+
+__all__ = [
+    'StochasticProcess',
+    'DiscreteMarkovChain'
+]
+
+class StochasticProcess(Basic):
+    """
+    Base class for all the stochastic process whether
+    discrete or continuous.
+
+    Parameters
+    ==========
+
+    sym: Symbol or string_types
+    state_space: Set
+        The state space of the stochastic process, by default S.Reals.
+        For discrete sets it is zero indexed.
+    """
+
+    is_Continuous = None
+    is_Discrete = None
+
+    def __new__(cls, sym, state_space=S.Reals):
+        if isinstance(sym, string_types):
+            sym = Symbol(sym)
+        if not isinstance(sym, Symbol):
+            raise TypeError("Name of stochastic process should be either only "
+                            "a string or Symbol.")
```

Comment:
Is it really necessary to accept both symbols and strings? If it is then you repeat the same code in many places. Can there not be a helper for this or can it be moved to a superclass?

Reply:
> Is it really necessary to accept both symbols and strings? 

Yes, to provide options to the user.

> Can there not be a helper for this or can it be moved to a superclass?

Sure, I will define a helper function. Thanks for the idea.